code/game/objects/items/stacks/stack_recipes.dm

```diff
@@ -703,3 +703,56 @@ var/list/datum/stack_recipe/ralloy_recipes = list (
 var/list/datum/stack_recipe/sand_recipes = list (
 	new/datum/stack_recipe("sandstone", /obj/item/stack/sheet/mineral/sandstone, 1, 1, 50),
 	)
+
+/* ========================================================================
+							FEATHER RECIPES
+======================================================================== */
+
+var/list/datum/stack_recipe/feather_recipes = list(
+    new/datum/stack_recipe/feather("feather wreath", /obj/item/mounted/frame/wreath/feather, 4, time = 50, one_per_turf = 1, on_floor = 1),
+	new /datum/stack_recipe/feather("feather coat", /obj/item/clothing/suit/feathercoat, 6, time = 50, one_per_turf = 1),
+	new /datum/stack_recipe/feather("feather vest", /obj/item/clothing/suit/feathervest, 4, time = 50, one_per_turf = 1),
+	new /datum/stack_recipe/feather("dreamcatcher", /obj/item/mounted/frame/dreamcatcher, 4, time = 50, one_per_turf = 1),
+	new /datum/stack_recipe/feather("pillow", /obj/item/weapon/pillow, 2, time = 30),
+	new /datum/stack_recipe/feather("quill pen", /obj/item/weapon/pen/quill, 1, time = 10),
+	new /datum/stack_recipe/feather("feather duster", /obj/item/weapon/featherduster, 2, time = 20),
+	new /datum/stack_recipe/feather("headdress", /obj/item/clothing/head/headdress, 3, time = 60),
+)
+
+// Feather-specific stack recipe for color inheritance
+/datum/stack_recipe/feather
+	parent_type = /datum/stack_recipe
+
+
+/datum/stack_recipe/feather/finish_building(var/mob/user, var/obj/item/stack/S, var/obj/R)
+	// All feather craftables inherit feather color and color name from the stack
+	if(istype(S, /obj/item/stack/feather))
+		var/obj/item/stack/feather/FS = S
+		if(istype(R, /obj/item/weapon/pillow))
+			var/obj/item/weapon/pillow/P = new /obj/item/weapon/pillow(R.loc, FS.feather_color, FS.feather_color_name)
+			qdel(R)
+			R = P
```

Comment:
This is a weird way to do this; just use a new var name here to avoid confusion since you're qdel'ing the previous instance of this in the previous line.